packages/core/components/Modal/MoveFileManifest/index.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from "react";
+import { useSelector } from "react-redux";
+
+import { ModalProps } from "..";
+import BaseModal from "../BaseModal";
+import { PrimaryButton } from "../../Buttons";
+import { interaction, selection } from "../../../state";
+import FileDetail from "../../../entity/FileDetail";
+import styles from "./MoveFileManifest.module.css";
+import FileSelection from "../../../entity/FileSelection";
+
+/**
+ * Modal overlay for displaying details of selected files for NAS cache operations.
+ */
+export default function MoveFileManifest({ onDismiss }: ModalProps) {
+    // const dispatch = useDispatch(); //TODO: add onMove functionality
+    const fileSelection = useSelector(
+        selection.selectors.getFileSelection,
+        FileSelection.selectionsAreEqual
+    );
+    const moveFileTarget = useSelector(interaction.selectors.getMoveFileTarget);
+
+    const [fileDetails, setFileDetails] = React.useState<FileDetail[]>([]);
+    const [totalSize, setTotalSize] = React.useState<number>(0);
+
+    React.useEffect(() => {
+        async function fetchDetails() {
+            const details = await fileSelection.fetchAllDetails();
+            setFileDetails(details);
+            const totalFileSize = details.reduce((acc, file) => acc + (file.size || 0), 0);
+            setTotalSize(totalFileSize);
+        }
+
+        fetchDetails();
+    }, [fileSelection]);
+
+    const onMove = () => {
+        console.log(
+            `Moving ${fileDetails.length} files ${
+                moveFileTarget === "ON_TO_NAS" ? "onto" : "off of"
+            } NAS.`
+        );
+        onDismiss();
+    };
+
+    const body = (
+        <div>
+            <p>Selected Files:</p>
+            <div className={styles.fileListContainer}>
+                <ul className={styles.fileList}>
+                    {fileDetails.map((file) => (
+                        <li key={file.id} className={styles.fileItem}>
```

Comment:
I think a table with the columns name & size would be better than this list for readability. Like so:
```
| File Name              | File Size  |
| my_file.txt             | 145 MB    |
| another_one.png | 206MB     |
```

Reply:
resolved in [77a52bd](https://github.com/AllenInstitute/biofile-finder/pull/311/commits/77a52bd92652b8a0f7f0eeb5ed43216f73ff3045)